README.md

```diff
@@ -20,12 +20,36 @@ Any exceptions to this are noted in the individual `README.md` files.
 
 ## Installation
 
-TBD
+The repository layout structure is as defined by [the specification](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/library-layout.md). The `api`
+directory includes header-only abstract classes and no-op implementations that comprise the OpenTelemetry API following
+[the
+specification](https://github.com/open-telemetry/opentelemetry-specification).
+The `sdk` folder is the reference implementation of the API.
+
+The library would be distributed in source-only format. No shared/static library or distribution package (rpm, deb etc) would be made available as part of Release. See [RELEASING.md](./RELEASING.md).
+
+This project supports `Bazel` and `CMake` builds.
+
+### Using CMake
+On most platforms with all dependencies installed, the following command will compile the source files, and install the header files:
 
-## Quick Start
+```sh
+cmake -Bcmake-out
```

Comment:
This may be a bit too generic. We need more detailed step-by-step instructions here, such as min cmake version, min compiler version, min required versions of Google Test. Just having generic `cmake` instruction is a good start. But may not necessarily be sufficient. For `ctest` - we can run this if built with tests as part of a single `cmake build` command.

Reply:
Sure, we can reference to "INSTALL.md" document, as you plan to document it separately ?

Reply:
I can cover this part. You can remove the build instructions for now.

Reply:
I have added reference to INSTALL.md under this section. I still feel it makes sense to keep generic installation steps as it given some idea to devs/users on how to build the library. I have already mentioned "with all dependencies specified". Regarding ctest, the default build using `cmake build` does build the tests , so these command should work out of box to run tests too.

Also have added the recommended deployment directory structure to be used by package manager. Let me know if this is ok.

Reply:
We should add a _Requirements_ section, either here on in the `INSTALL.md`. I think the LightStep OpenTracing C++ project does a good job in describing requirements as well as detailed build instructions: https://github.com/lightstep/lightstep-tracer-cpp#requirements